runtime/src/bank.rs

```diff
@@ -6285,6 +6272,15 @@ impl Bank {
         self.builtin_feature_transitions = Arc::new(builtins.feature_transitions);
 
         self.apply_feature_activations(true, debug_do_not_add_builtins);
+
+        if self
+            .feature_set
+            .is_active(&feature_set::cap_accounts_data_len::id())
+        {
+            self.cost_tracker = RwLock::new(CostTracker::new_with_account_data_size_limit(Some(
+                MAX_ACCOUNTS_DATA_LEN.saturating_sub(self.accounts_data_size_initial),
+            )));
+        }
```

Comment:
I moved this code into `finish_init()` so that it can be shared by `new_from_fields()` and `new_with_paths()`.